src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/reader/ClusterDetailsEventProcessor.java

```diff
@@ -41,7 +42,15 @@
    */
   private volatile ImmutableList<NodeDetails> nodesDetails = null;
 
-  public ClusterDetailsEventProcessor() {}
+  private final ConfigOverridesApplier overridesApplier;
+
+  public ClusterDetailsEventProcessor() {
+    this(new ConfigOverridesApplier());
+  }
+
+  public ClusterDetailsEventProcessor(final ConfigOverridesApplier overridesApplier) {
```

Comment:
Maybe we don't need to pass `ConfigOverridesApplier` as a constructor. No state is kept in it. It reads the config and applies it to the `rca.conf`. Maybe we can just use it as a helper static class ?

Reply:
It does keep the `lastAppliedTimestamp` as an instance variable and uses that to check if the config overrides string received from the `ClusterDetailsEventProcessor`. Making it a static helper will make all the consumers see the same value of last applied time(in an IT for example).